test/cases/test_pgsql.py

```
@@ -0,0 +1,422 @@
+                            f"Row '{tag}' did not replicate to subscriber within 5 seconds")
+        finally:
+            # Cleanup on publisher — DELETE also replicates
+            self._psql("postgres", "testdb_pg", "pguser",
+                       f"DELETE FROM items_pg WHERE name = '{tag}'")
+
+    def test_delete_replicates_to_subscriber(self):
+        """A row deleted on the publisher must be removed from the subscriber."""
+        tag = f"repl-del-{int(time.time())}"
+        # Insert and wait for replication
+        self._psql("postgres", "testdb_pg", "pguser",
+                   f"INSERT INTO items_pg (name, val) VALUES ('{tag}', 99)")
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)